File: we-care-app-frontend/src/app/common-components/models/doctor-clinic-schedule-dto.ts
import { ScheduleDTO } from "./schedule-dto";

export interface DoctorClinicScheduleDTO {

    doctorId: number,
    userId: number,
    firstName: string,
    lastName: string,
    briefIntroduction: string,
    joiningDate: string,
    doctorImageURL: string,
    totalRating: number | null,
    averageRating: number | null,
    specialityId: number,
    specialityName:string,
    fees: number,
    lastUpdated: string,
    clinicState: string,
    clinicCity: string,
    address: string,
    clinicMobile: string,
    clinicId: number,
    scheduleDTOs: ScheduleDTO [];
}

File: we-care-app-backend/src/main/java/clinicmangement/com/WeCare/DTO/DoctorClinicScheduleDTO.java
package clinicmangement.com.WeCare.DTO;

import clinicmangement.com.WeCare.enums.ScheduleStatus;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class DoctorClinicScheduleDTO {
    //doctor
    private Integer doctorId;
    private Integer userId;
    private String firstName;
    private String lastName;
    private String briefIntroduction;
    private LocalDateTime joiningDate;
    private String doctorImageURL;
    private Double totalRating;
    private Double averageRating;
    private Integer specialityId;
    private String specialityName;
    private Integer fees;
    private Long lastUpdated;
    //clinic
    private Integer clinicId;
    private String clinicState;
    private String clinicCity;
    private String address;
    private String clinicMobile;
    //schedule
    private List<ScheduleDTORead> scheduleDTOs;
    private ScheduleStatus scheduleStatus;
}
